src/highlight/highlightUtil.ts

```diff
@@ -18,6 +19,47 @@ export function getHighlightArea(shapes: Rect[]): number {
     return shapes.reduce((area, { height, width }) => area + height * width, 0);
 }
 
+export function getRange(): Range | null {
+    const selection = window.getSelection();
+    if (!selection || selection.type !== 'Range' || selection.isCollapsed) {
+        return null;
+    }
+
+    return selection.getRangeAt(0);
+}
+
+export const domRectToMini = ({ height, width, x, y }: DOMRect): DOMRectMini => ({
+    height,
+    width,
+    x,
+    y,
+});
+
+export function getSelectionItem(): {
+    boundingRect: DOMRectMini;
+    location: number;
+    rects: Array<DOMRectMini>;
+} | null {
+    const range = getRange();
+
+    if (!range) {
+        return null;
+    }
+
+    const startPage = getPageNumber(range.startContainer as Element);
+    const endPage = getPageNumber(range.endContainer as Element);
+
+    if (!startPage || !endPage || startPage !== endPage) {
+        return null;
+    }
+
+    return {
+        boundingRect: domRectToMini(range.getBoundingClientRect()),
+        location: endPage,
+        rects: Array.from(range.getClientRects()).map(domRectToMini),
```

Comment:
Should we be storing the raw bounding client rects? Or should we be determining the % values wrt the content that we will eventually need to render & save to the API?

Reply:
I think we want to separate selection store from creator store. We can reformat the raw data when it is promoted to staged.